src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -1,82 +1,21 @@
 import { promises as fsPromises } from 'fs';
 import * as mime from 'mime-types';
 import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
-import { getLoggerFor } from '../../logging/LogUtil';
 import { APPLICATION_OCTET_STREAM, TEXT_TURTLE } from '../../util/ContentTypes';
 import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
-import {
-  encodeUriPathComponents,
-  ensureTrailingSlash,
-  isContainerIdentifier,
-  joinFilePath,
-  normalizeFilePath,
-  trimTrailingSlashes,
-} from '../../util/PathUtil';
-import type { FileIdentifierMapper, FileIdentifierMapperFactory, ResourceLink } from './FileIdentifierMapper';
-import { getAbsolutePath, getRelativePath, validateRelativePath } from './MapperUtil';
-
-export interface ResourcePath {
-
-  /**
-   * The path of the container.
-   */
-  containerPath: string;
-
-  /**
-   * The document name.
-   */
-  documentName?: string;
-}
-
-/**
- * A mapper that stores the content-type of resources in the file path extension.
- * In case the extension of the identifier does not correspond to the correct content-type,
- * a new extension will be appended (with a `$` in front of it).
- * E.g. if the path is `input.ttl` with content-type `text/plain`, the path would actually be `input.ttl$.txt`.
- * This new extension is stripped again when generating an identifier.
- *
- * Warning: Since this mapper iterates over all files in the requested directory,
- * it can experience performance issues over directories with a huge number of files (10.000+).
- * For typical directory structures, the performance of this mapper should be sufficient.
- * @see https://github.com/solid/community-server/issues/333
- */
-export class ExtensionBasedMapper implements FileIdentifierMapper {
-  protected readonly logger = getLoggerFor(this);
-
-  private readonly baseRequestURI: string;
-  private readonly rootFilepath: string;
+import { joinFilePath } from '../../util/PathUtil';
+import type { FileIdentifierMapperFactory, ResourceLink } from './FileIdentifierMapper';
+import { FixedContentTypeMapper } from './FixedContentTypeMapper';
+export class ExtensionBasedMapper extends FixedContentTypeMapper {
```

Comment:
From a hierarcy point of view this feels wrong, since now we're saying that an `ExtensionBasedMapper` is a `FixedContentTypeMapper` mapper. But a `FixedContentTypeMapper` is a specific kind of mapper that only supports 1 specific content-type. So I think it would make more sense to have something like a `BaseFileIdentifierMapper` that both of them extend.

Reply:
Yeah; thought about that too, but thought it might make things more complicated. Will do.